subsystem/cluster_test.go

```diff
@@ -212,6 +235,9 @@ func waitForClusterState(ctx context.Context, clusterID strfmt.UUID, state strin
 
 		if success {
 			successInRow++
+			// Check if the lastState has already skipped the expected state
+		} else if stateIndex := indexOf(state, clusterValidStatuses[:]); stateIndex != -1 && indexOf(lastState, clusterValidStatuses[:]) > stateIndex {
```

Comment:
It is a more generic way to verify if a current status has been already changed and the subsystem test didn't manage to "catch" that moment, especially in an asynchronous environment. So if it was skipped into a higher state - It is fine.
We can change **every place** that calls `InstallCluster`, but it checks a different thing - the returned value by `InstallCluster` instead of the cluster status on `GetCluster` (which might never get checked and might not transformed neither).

Reply:
My main issue that if someone will add a state they will not go into this list and know to update the right order, they will probably just add it in the end.

Reply:
You can say the same on the [host stages](https://github.com/openshift/assisted-service/blob/master/internal/host/host.go#L30)
We don't have any static assertion for including all of the elements. (if that's possible - it might be a solution)

If a contributor creates a new stage/status and doesn't update the list - It won't affect anything.
If he is updating the list - he should know how to and that's also why we require a code review.

Reply:
There is a difference between production code and test code.